src/main.rs

```diff
@@ -130,144 +136,144 @@ fn main() {
 
     let toml_str = &fs::read_to_string("info.toml").unwrap();
     let exercises = toml::from_str::<ExerciseList>(toml_str).unwrap().exercises;
-    let verbose = matches.is_present("nocapture");
+    let verbose = args.nocapture;
 
-    // Handle the list command
-    if let Some(list_m) = matches.subcommand_matches("list") {
-        if ["paths", "names"].iter().all(|k| !list_m.is_present(k)) {
-            println!("{:<17}\t{:<46}\t{:<7}", "Name", "Path", "Status");
-        }
-        let filters = list_m.value_of("filter").unwrap_or_default().to_lowercase();
-        let mut exercises_done: u16 = 0;
-        exercises.iter().for_each(|e| {
-            let fname = format!("{}", e.path.display());
-            let filter_cond = filters
-                .split(',')
-                .filter(|f| !f.trim().is_empty())
-                .any(|f| e.name.contains(&f) || fname.contains(&f));
-            let status = if e.looks_done() {
-                exercises_done = exercises_done + 1;
-                "Done"
-            } else {
-                "Pending"
-            };
-            let solve_cond = {
-                (e.looks_done() && list_m.is_present("solved"))
-                    || (!e.looks_done() && list_m.is_present("unsolved"))
-                    || (!list_m.is_present("solved") && !list_m.is_present("unsolved"))
-            };
-            if solve_cond && (filter_cond || !list_m.is_present("filter")) {
-                let line = if list_m.is_present("paths") {
-                    format!("{}\n", fname)
-                } else if list_m.is_present("names") {
-                    format!("{}\n", e.name)
+    let command = args.nested.unwrap_or_else(|| {
+        let text = fs::read_to_string("default_out.txt").unwrap();
+        println!("{}", text);
+        std::process::exit(0);
+    });
+    match command {
+        Subcommands::List(subargs) => {
+            if !subargs.paths && !subargs.names {
+                println!("{:<17}\t{:<46}\t{:<7}", "Name", "Path", "Status");
+            }
+            let mut exercises_done: u16 = 0;
+            let filters = subargs.filter.clone().unwrap_or_default().to_lowercase();
+            exercises.iter().for_each(|e| {
+                let fname = format!("{}", e.path.display());
+                let filter_cond = filters
+                    .split(',')
+                    .filter(|f| !f.trim().is_empty())
+                    .any(|f| e.name.contains(&f) || fname.contains(&f));
+                let status = if e.looks_done() {
+                    exercises_done = exercises_done + 1;
```

Comment:
Carryover from previous commit but this can be shortened.
https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
```suggestion
                    exercises_done += 1;
```